LayoutTests/css3/self-alignment/block-justify-self-rtl.html

```diff
@@ -0,0 +1,213 @@
+      <div class="actual" style="justify-self: auto;">actual</div>
+    </div>
+  </dd>
+
+  <dt>normal acts as default layout</dt>
+  <dd>
+    <div class="test">
+      <div class="expected">expected</div>
+      <div class="actual" style="justify-self: normal">actual</div>
+    </div>
+  </dd>
+
+  <dt>stretches if auto width and !auto margins</dt>
+  <dd>
+    <div class="test">
+      <div class="expected" style="width: auto;">expected</div>
+      <div class="actual" style="justify-self: stretch; width: auto; margin: 0;">actual</div>
+    </div>
+  </dd>
+
+  <dt>stretch = flex-start if fixed width</dt>
+  <dd>
+    <div class="test">
+      <div class="expected">expected</div>
+      <div class="actual" style="justify-self: stretch; margin: 0;">actual</div>
+    </div>
+  </dd>
+
+  <dt>stretch = flex-start if auto margins</dt>
+  <dd>
+    <div class="test">
+      <div class="expected" style="width: auto;">expected</div>
+      <div class="actual" style="width: auto; justify-self: stretch;">actual</div>
+    </div>
+  </dd>
+
+  <dt>stretch respects max-width</dt>
+  <dd>
+    <div class="test">
+      <div class="expected" style="width: 150px;">expected</div>
+      <div class="actual" style="justify-self: stretch; width:auto; max-width: 150px;">actual</div>
+    </div>
+  </dd>
+
+  <dt>center</dt>
+  <dd>
+    <div class="test">
+      <div class="expected" style="margin: auto;">expected</div>
+      <div class="actual" style="justify-self: center;">actual</div>
+    </div>
+  </dd>
+
+  <dt>left</dt>
+  <dd>
+    <div class="test">
+      <div class="expected" style="margin-right: 150px;">expected</div>
+      <div class="actual" style="justify-self: left;">actual</div>
+    </div>
+  </dd>
+
+  <dt>right</dt>
+  <dd>
+    <div class="test">
+      <div class="expected" style="margin-left: 150px;">expected</div>
+      <div class="actual" style="justify-self: right;">actual</div>
+    </div>
+  </dd>
+
+  <dt>start</dt>
+  <dd>
+    <div class="test">
+      <div class="expected">expected</div>
+      <div class="actual" style="justify-self: start;">actual</div>
+    </div>
+  </dd>
+
+  <dt>end</dt>
+  <dd>
+    <div class="test">
+      <div class="expected" style="margin-right: 150px;">expected</div>
+      <div class="actual" style="justify-self: end;">actual</div>
+    </div>
+  </dd>
+
+  <dt>flex-start</dt>
+  <dd>
+    <div class="test">
+      <div class="expected">expected</div>
+      <div class="actual" style="justify-self: flex-start;">actual</div>
+    </div>
+  </dd>
+
+  <dt>flex-end</dt>
+  <dd>
+    <div class="test">
+      <div class="expected" style="margin-right: 150px;">expected</div>
+      <div class="actual" style="justify-self: flex-end;">actual</div>
+    </div>
+  </dd>
+
+  <dt>self-start</dt>
+  <dd>
+    <div class="test">
+      <div class="expected" style="direction: ltr; margin-right: 150px;">expected</div>
+      <div class="actual" style="justify-self: self-start; direction: ltr">actual</div>
```

Comment:
Using a conflicting direction here is important, but actually you need to do it for the other tests as well, so that it's clear in those cases that you're using the containing block direction and not the item.